### PR TITLE
✨Add rich click

### PR DESCRIPTION
**Pull request by mayankpatibandla**
#### Summary:
Adds rich click

#### Motivation:
Create better formatting for the CLI commands

#### Test Plan:
- [x] Windows
- [x] Linux (SSH)
- [x] M2 Mac
- [x] Intel Mac

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 click>=6,<7
```
Comment (ayushuk) [inline requirements.txt]: after some testing it looks like we need to bump to click 8

**Review by ayushuk — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
-click>=6,<7
+click>=8
```
Comment (ayushuk) [inline requirements.txt:1]: Lets not bump all the way to 8, I know the minimum requirement for rich click is >=7 so lets stick with that now and bump it later on.

File: requirements.txt [hunk 3]
```diff
@@ -13,4 +14,4 @@ scan-build==2.0.13
 sentry-sdk
 observable
 pypng==0.0.20
-pyinstaller
\ No newline at end of file
+pyinstaller
```
Comment (ayushuk) [inline requirements.txt:17]: I forget, why was this added?

Comment (mayankpatibandla) [reply requirements.txt:17]: Not sure why that shows up on the diff, I'm pretty sure that line is unchanged.

**Comment by ayushuk** [comment]
I was experimenting one last time and it seems some of the help text is missing and for some reason hidden commands are shown on mac. (commands like `build-compile-commands` are marked hidden)

Screenshot 2023-12-22 at 8 44 57 PM

All I did was switch to the branch, `pip install -r "requirements.txt"` and ran pros.

**Comment by ayushuk** [comment]
For some reason this isn't an issue on my Ubuntu machine and both have the same version of click (8.1.7).

Expected behavior:
Screenshot 2023-12-22 at 8 54 19 PM

**Comment by mayankpatibandla** [comment]
Works fine on Windows 10, click 8.1.7. No idea what could cause this discrepancy.
image

**Comment by ayushuk** [comment]
I'll investigate this more when I get the chance tomorrow. Probably something small and it could be related to my specific setup.

**Review by ayushuk — APPROVED** [review]
The issue was related to my setup. Works for me on MacOS, if I'm not wrong then I think we've tested pretty much everywhere now so we should be good to merge.
